File: src/com/gmail/marcosav2010/communicator/module/fth/FileSendInfo.java
package com.gmail.marcosav2010.communicator.module.fth;

import java.nio.file.Path;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
public class FileSendInfo {

	@Getter
	private final Path path;
	@Getter
	private final int size;
	@Getter
	private final int blocks;
	@Getter
	private final int blockSize;
	@Getter
	private int fileID = -1;

	public void setFileID(int fileID) {
		if (this.fileID == -1)
			this.fileID = fileID;
		else
			throw new IllegalStateException("ID is already set.");
	}

	public String getFileName() {
		return path.getFileName().toString();
	}
}

File: modules/file-transfer/src/main/java/com/gmail/marcosav2010/fth/FileSendInfo.kt
package com.gmail.marcosav2010.fth

import java.nio.file.Path

class FileSendInfo internal constructor(
        val path: Path,
        val size: Int = 0,
        val blocks: Int = 0,
        val blockSize: Int = 0
) {

    var fileID = -1
        private set

    val fileName: String
        get() = path.fileName.toString()

    fun setFileID(fileID: Int) {
        if (this.fileID == -1)
            this.fileID = fileID
        else
            throw IllegalStateException("ID is already set.")
    }
}